nano/secure/ledger.cpp

```diff
@@ -466,12 +446,12 @@ void ledger_processor::change_block (nano::change_block & block_a)
 			result.code = block_a.valid_predecessor (*previous) ? nano::process_result::progress : nano::process_result::block_position;
 			if (result.code == nano::process_result::progress)
 			{
-				auto account (ledger.store.frontier.get (transaction, block_a.hashables.previous));
-				result.code = account.is_zero () ? nano::process_result::fork : nano::process_result::progress;
+				auto account = ledger.account (*previous);
+				auto info = ledger.account_info (transaction, account);
+				debug_assert (info);
```

Comment:
From std::optional
> The behavior is undefined if *this does not contain a value.

Let's use `release_asserts` in such a critical piece of code (almost everywhere in ledger.cpp). There's no recovery codepath, so this should completely crash.

Reply:
Agreed, I will raise a separate PR to replace the debug_asserts in the ledger_processor with release_asserts.